iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -322,81 +326,83 @@ private Task<CommandResponse> HandleMaxMinReportCommandAsync(CommandRequest requ
+            /*string serializedProperties = ((JObject)desiredProperties.Value).ToString();
+            TemperatureRange temeratureRangeDesired = s_payloadConvention.PayloadSerializer.DeserializeToType<TemperatureRange>(serializedProperties);*/
+
+            TemperatureRange temperatureRangeDesired = writableProperties.GetValue<TemperatureRange>(propertyName);
 
             var temperatureUpdateResponse = new CustomWritablePropertyResponse(
-                temeratureRangeDesired,
+                temperatureRangeDesired,
                 (int)StatusCode.Completed,
-                version,
+                writableProperties.Version,
                 "The operation completed successfully.");
 
             // A property collection with WritablePropertyResponse needs to be always vserialized with NewtonSoft
             // => we cannot merge it with user defined serialization settings.
-            var propertyPatch = new PropertyCollection();
-            propertyPatch.Add(propertyName, temperatureUpdateResponse);
+            var propertyPatch = new PropertyCollection(s_payloadConvention)
+            {
+                [propertyName] = temperatureUpdateResponse,
+            };
 
-            await _deviceClient.UpdatePropertiesAsync(propertyPatch, cancellationToken: cancellationToken);
+            await _deviceClient.UpdatePropertiesAsync(propertyPatch, s_cancellationToken);
             _logger.LogDebug($"Property: Update - {{\"{propertyName}\" is complete.");
         }
 
         // The desired property update callback, which receives the target temperature as a desired property update,
         // and updates the current temperature value over telemetry and property update.
         // This callback is invoked for all updates received for the two associated components.
-        private async Task TargetTemperatureUpdateCallbackAsync(KeyValuePair<string, object> desiredProperties, long version, object userContext, string dispatcherKey)
+        private async Task TargetTemperatureUpdateCallbackAsync(PropertyCollection writableProperties, object userContext, string dispatcherKey)
         {
             string componentName = dispatcherKey;
-            var cancellationToken = (CancellationToken)userContext;
 
             // The component has a single writable property, so the property name is a const here.
             // Otherwise this would have been a switch-case.
             const string propertyName = "targetTemperature";
 
             // PropertyCollection.Value is now always JObject (since we create PropertyCollection from TwinCollection).
             // This implementation detail will need to be addressed.
-            bool targetTempUpdateReceived = ((Dictionary<string, object>)desiredProperties.Value).ContainsKey(propertyName);
+            bool targetTempUpdateReceived = ((JsonElement)writableProperties[componentName]).TryGetProperty(propertyName, out JsonElement targetTemperatureJson);
```

Comment:
same here, very difficult and unintuitive to retrieve the value of a component-level property